src/blocked.c

```diff
@@ -240,6 +240,8 @@ void replyToBlockedClientTimedOut(client *c) {
         addReplyLongLong(c,replicationCountAOFAcksByOffset(c->bstate.reploffset));
     } else if (c->bstate.btype == BLOCKED_MODULE) {
         moduleBlockedClientTimedOut(c);
+    } else if(c->bstate.btype == BLOCKED_DEALY){
+        c->duration = 0;
```

Comment:
why do we need to reset `c->duration`?
p.s. we're obviously missing some tests,
let's add them after we agree on the main details.